src/NServiceBus.Core/Receiving/ReceiveComponent.cs

```diff
@@ -69,16 +90,18 @@ public static Configuration PrepareConfiguration(Settings settings, TransportCom
                 settings.PipelineCompletedSubscribers ?? new Notification<ReceivePipelineCompleted>(),
                 isSendOnlyEndpoint,
                 settings.ExecuteTheseHandlersFirst,
-                settings.MessageHandlerRegistry);
+                settings.MessageHandlerRegistry,
+                transportInfrastructure,
```

Comment:
not really sure whether we should pass it around within the configuration or pass it again the ReceiveComponent.Initialize instead?

Reply:
Initialize() will be called by EndpointCreator that will have the ReceivingConfiguration with all the things needed for the ReceiveComponent to work. While it adds a bit of ceremony, personally I like the cutoff from the Settings and working with Configuration only.